sotodlib/hwp/hwp.py

```diff
@@ -454,42 +481,48 @@ def estimate_sigma_tod(signal, hwp_angle):
     return hwpss_sigma_tod
 
 
-def subtract_hwpss(aman, signal=None, hwpss_template=None,
-                   subtract_name='hwpss_remove'):
+def subtract_hwpss(aman, signal_name='signal', hwpss_template_name='hwpss_model',
+                   subtract_name='hwpss_remove', in_place=False, remove_template=True):
     """
     Subtract the half-wave plate synchronous signal (HWPSS) template from the
     signal in the given axis manager.
 
     Parameters
     ----------
     aman : AxisManager
-        The axis manager containing the signal to which the HWPSS template will
-        be applied.
-    signal : ndarray, optional
-        The signal to which the HWPSS template will be applied. If `signal` is
-        None (default), the signal contained in the axis manager will be used.
-    hwpss_template : ndarray, optional
-        The HWPSS template to be subtracted from the signal. If `hwpss_template`
-        is None (default), the HWPSS template stored in the axis manager under
-        the key 'hwpss_extract' will be used.
+        The axis manager containing the signal and the HWPSS template.
+    signal_name : str, optional
+        The name of the field in the axis manager containing the signal to be processed.
+        Defaults to 'signal'.
+    hwpss_template_name : str, optional
+        The name of the field in the axis manager containing the HWPSS template.
+        Defaults to 'hwpss_model'.
```

Comment:
The change over to `..._name` is no doubt to give some calling entity an easier job of things.  But it's much less general than accepting an array, because now the hwpss_template definitely needs to be in the AxisManager. 

In other parts of the code, such arguments are handled like this:
- If it's None, use the default.
- If it's a string, look up that string in the axisman.
- If it's an array, use the array.

That supports both modes of calling.  The logic can be written up into a helper function (and has, all over the place, I think), if you have multiple args to handle this way (which you do).

So I am hesitant to pass this breaking interface change that also is a bit of a regression.  I'd rather add additional logic to handle when "hwpss_template" is passed as a string, without removing the old interface.

Reply:
Ok I think I've addressed this now up to the fact that we want to fix the core issue in the tod_ops filters module.